### ISSUE TITLE
httpFancyWriter.ReadFrom not recording bytes written

### ISSUE THREAD

**Issue by mrbanzai** [go-chi/chi#233]
In debugging an issue with the `resp_bytes_length` output from the request logger, I stumbled upon this block of code (which gets triggered by `io.Copy`): [https://github.com/go-chi/chi/blob/4c5a584b324b74af3e9cfeaf6265d14ae2fdfc99/middleware/wrap_writer.go#L112-L119](url)

It seems that this is not updating the bytes written on the `basicWriter`.  Is this intentional, or simply an oversight?

**Comment by pkieltyka** [comment]
hey @mrbanzai what kind of issue are you running into with `resp_bytes_length`? have you tried the logging example? would you be able to provide some sample code for your issue?

**Comment by mrbanzai** [comment]
I have tried the logging example.  For most operations, it works without issue.  I, however, encountered an issue when streaming a file through `cipher.StreamWriter` via `io.Copy` (for encrypting content to the response writer).  By tracing through the Go 1.8 source, I noticed that it uses `ReadFrom` if the writer supports it, which led me to the section of code referenced above.  I was able to correct my issue by altering the code to update the `bytes` value on the `basicWriter` within the `httpFancyWriter.ReadFrom` (https://github.com/mrbanzai/chi/commit/9936a163a79d1ea4c634554c97bf38833fb403cf).
